pkg/routeagent_driver/handlers/mtu/mtuhandler.go

```diff
@@ -274,29 +276,54 @@ func (h *mtuHandler) Stop(uninstall bool) error {
 	return nil
 }
 
-func (h *mtuHandler) forceMssClamping(endpoint *submV1.Endpoint) error {
-	defaultHostIface, err := netlinkAPI.GetDefaultGatewayInterface()
-	if err != nil {
-		return errors.Wrapf(err, "Unable to find the default interface on host")
-	}
+func (h *mtuHandler) forceMssClamping(endpoint *submV1.Endpoint, usrForceMssVal int) error {
+	mssValueSrc := "user"
+	if usrForceMssVal == 0 {
+		defaultHostIface, err := netlinkAPI.GetDefaultGatewayInterface()
+		if err != nil {
+			return errors.Wrapf(err, "Unable to find the default interface on host")
+		}
 
-	overHeadSize := maxIpsecOverhead
-	if endpoint.Spec.Backend == vxlan.CableDriverName {
-		overHeadSize = vxlan.VxlanOverhead
-	}
+		overHeadSize := maxIpsecOverhead
+		if endpoint.Spec.Backend == vxlan.CableDriverName {
+			overHeadSize = vxlan.VxlanOverhead
+		}
 
-	klog.Infof("forceMssClamping to: IF_MTU(%d)-Overhead(%d)=%d",
-		defaultHostIface.MTU, overHeadSize, (defaultHostIface.MTU - overHeadSize))
+		usrForceMssVal = defaultHostIface.MTU - overHeadSize
+		mssValueSrc = "default"
+	}
 
+	klog.Infof("forceMssClamping to: %d (%s) ", usrForceMssVal, mssValueSrc)
 	ruleSpecSource := []string{
 		"-m", "set", "--match-set", constants.LocalCIDRIPSet, "src", "-m", "set", "--match-set",
 		constants.RemoteCIDRIPSet, "dst", "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j", "TCPMSS",
-		"--set-mss", strconv.Itoa(defaultHostIface.MTU - overHeadSize),
+		"--set-mss", strconv.Itoa(usrForceMssVal),
 	}
 	ruleSpecDest := []string{
 		"-m", "set", "--match-set", constants.RemoteCIDRIPSet, "src", "-m", "set", "--match-set",
 		constants.LocalCIDRIPSet, "dst", "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j", "TCPMSS",
-		"--set-mss", strconv.Itoa(defaultHostIface.MTU - overHeadSize),
+		"--set-mss", strconv.Itoa(usrForceMssVal),
+	}
+
+	rules, err := h.ipt.List(constants.MangleTable, constants.SmPostRoutingChain)
+	if err != nil {
+		return errors.Wrapf(err, "error listing the rules in %s chain", constants.SmPostRoutingChain)
+	}
+
+	isPresent := false
+
+	for _, rule := range rules {
+		if strings.Contains(rule, strings.Join(ruleSpecSource, " ")) || strings.Contains(rule, strings.Join(ruleSpecDest, " ")) {
+			isPresent = true
+			break
+		}
+	}
+
+	if !isPresent {
+		if err := h.ipt.ClearChain(constants.MangleTable, constants.SmPostRoutingChain); err != nil {
+			klog.Errorf("Error flushing iptables chain %q of %q table: %v", constants.SmPostRoutingChain,
+				constants.MangleTable, err)
```

Comment:
Should we return an error here?

Reply:
Hmm, maybe we can log warning or something like that and create the new iptables rules

Reply:
Ack, warning should be fine. We should also add a check to see that there are indeed some rules in the chain otherwise this warning will be displayed on a fresh setup as well. 

i.e., if !isPresent && len(rules) > 0

Reply:
Done